### PR TITLE
[DOC] Add a tutorial for the phred score alphabet

### PR DESCRIPTION
**Pull request by Irallia**
This is the last part, and so it 
Resolves https://github.com/seqan/product_backlog/issues/266

### CODE REVIEW

File: doc/tutorial/alphabet/index.md [hunk 1]
```diff
@@ -171,6 +171,26 @@ alphabet, or nucleotide / amino acid and structure alphabet that stores both inf
...
+seqan3::qualified. It represents the cross product between a nucleotide and a quality alphabet and is the go-to choice
+when compression is of interest.
+The following combinations still fit into a single byte:
+ - <code>seqan3::qualified<seqan3::dna4, seqan3::phred42></code> (alphabet size: 4 x 42 = 168)
+ - <code>seqan3::qualified<seqan3::dna4, seqan3::phred63></code> (alphabet size: 4 x 63 = 252)
+ - <code>seqan3::qualified<seqan3::dna5, seqan3::phred42></code> (alphabet size: 4 x 42 = 210)
+
+Using \ref seqan3::qualified can half the storage usage compared to storing qualities and nucleotides separately.
+
+While we used DNA alphabets in this example, the same properties hold true for RNA alphabets.
```
Comment (joergi-w) [inline doc/tutorial/alphabet/index.md:190]:
```suggestion
While we use DNA alphabets in this example, the same properties hold true for RNA alphabets.
```

**Review by joergi-w — APPROVED** [review]
Thank you, looks good! 
Just a tiny suggestion, which you can incorporate with the second review:

File: doc/tutorial/alphabet/alphabet_main.cpp [hunk 2]
```diff
@@ -76,6 +78,22 @@ int main ()
     assert(eq && neq && geq && gt && seq && st);
     assert(some_nucl == "AGT"_dna4);
 
+//! [phred]
+    using seqan3::operator""_dna4;
+    using seqan3::operator""_phred42;
+
+    seqan3::phred42 phred;
+    phred.assign_rank(2); // wrapper for assign_phred(2)
```
Comment (marehr) [inline doc/tutorial/alphabet/alphabet_main.cpp:86]:
```suggestion
    phred.assign_phred(2);
```

File: doc/tutorial/alphabet/index.md [hunk 3]
```diff
@@ -171,6 +171,26 @@ alphabet, or nucleotide / amino acid and structure alphabet that stores both inf
...
+sequencing. A Phred quality score is assigned to each nucleotide base call and in the FASTQ format encoded as ASCII
+characters alongside the read sequences.
+
+Therefore, it stands to reason to combine both alphabets into a new data structure. In SeqAn, this can be done with
+seqan3::qualified. It represents the cross product between a nucleotide and a quality alphabet and is the go-to choice
+when compression is of interest.
+The following combinations still fit into a single byte:
+ - <code>seqan3::qualified<seqan3::dna4, seqan3::phred42></code> (alphabet size: 4 x 42 = 168)
+ - <code>seqan3::qualified<seqan3::dna4, seqan3::phred63></code> (alphabet size: 4 x 63 = 252)
+ - <code>seqan3::qualified<seqan3::dna5, seqan3::phred42></code> (alphabet size: 4 x 42 = 210)
```
Comment (marehr) [inline doc/tutorial/alphabet/index.md:186]:
```suggestion
 - `seqan3::qualified<seqan3::dna4, seqan3::phred42>` (alphabet size: 4 x 42 = 168)
 - `seqan3::qualified<seqan3::dna4, seqan3::phred63>` (alphabet size: 4 x 63 = 252)
 - `seqan3::qualified<seqan3::dna5, seqan3::phred42>` (alphabet size: 4 x 42 = 210)
```

Is this the same?

File: doc/tutorial/alphabet/index.md (hunk 1 above)
Comment (marehr) [reply doc/tutorial/alphabet/index.md:190]: Open

File: doc/tutorial/alphabet/index.md [hunk 4]
```diff
@@ -171,6 +171,26 @@ alphabet, or nucleotide / amino acid and structure alphabet that stores both inf
...
+
+Using \ref seqan3::qualified can half the storage usage compared to storing qualities and nucleotides separately.
```
Comment (marehr) [inline doc/tutorial/alphabet/index.md:188]:
```suggestion
Using seqan3::qualified can half the storage usage compared to storing qualities and nucleotides separately.
```

Is the ref necessary?

File: test/snippet/alphabet/quality/phred42.cpp [hunk 5]
```diff
@@ -11,7 +11,7 @@ int main()
     seqan3::phred42 phred;
     phred.assign_rank(2); // wrapper for assign_phred(2)
```
Comment (marehr) [inline test/snippet/alphabet/quality/phred42.cpp:12]:
```suggestion
    phred.assign_phred(2);
```

File: test/snippet/alphabet/quality/phred63.cpp [hunk 6]
```diff
@@ -11,7 +11,7 @@ int main()
     seqan3::phred63 phred;
     phred.assign_rank(2); // wrapper for assign_phred(2)
```
Comment (marehr) [inline test/snippet/alphabet/quality/phred63.cpp:12]:
```suggestion
    phred.assign_phred(2);
```

File: doc/tutorial/alphabet/index.md [hunk 7]
```diff
@@ -171,6 +171,26 @@ alphabet, or nucleotide / amino acid and structure alphabet that stores both inf
...
+### Phred Score
+
+The Phred quality score is the \ref quality alphabet for nucleotides. It is generated automatically during DNA
+sequencing. A Phred quality score is assigned to each nucleotide base call and in the FASTQ format encoded as ASCII
+characters alongside the read sequences.
+
+Therefore, it stands to reason to combine both alphabets into a new data structure. In SeqAn, this can be done with
+seqan3::qualified. It represents the cross product between a nucleotide and a quality alphabet and is the go-to choice
+when compression is of interest.
+The following combinations still fit into a single byte:
```
Comment (marehr) [inline doc/tutorial/alphabet/index.md:183]:
```suggestion
when compression is of interest.

The following combinations still fit into a single byte (which can hold up-to 256 different values):
```

**Review by marehr — COMMENTED** [review]
Thank you for the PR, I think the PR does not resolve seqan/product_backlog#266. I wrote you some things I encountered in your gitter chat that should be fixed before resolving it.

**Comment by Irallia** [comment]
I will rebase and clean up after your confirmation.

File: doc/tutorial/alphabet/index.md (hunk 4 above)
Comment (marehr) [reply doc/tutorial/alphabet/index.md:189]: And did this work?

File: include/seqan3/alphabet/quality/phred42.hpp [hunk 8]
```diff
@@ -39,7 +39,7 @@ namespace seqan3
  *
  * \include test/snippet/alphabet/quality/phred42.cpp
  *
- * \see quality
+ * \see quality, it contains a comprehensive overview over all phred implementations and what a Phred score represents.
```
Comment (marehr) [inline include/seqan3/alphabet/quality/phred42.hpp:42]:
```suggestion
 * \see quality, it contains a comprehensive overview over all Phred implementations and what a Phred score represents.
```

File: include/seqan3/alphabet/quality/phred63.hpp [hunk 9]
```diff
@@ -39,7 +39,7 @@ namespace seqan3
  *
  * \include test/snippet/alphabet/quality/phred63.cpp
  *
- * \see quality
+ * \see quality, it contains a comprehensive overview over all phred implementations and what a Phred score represents.
```
Comment (marehr) [inline include/seqan3/alphabet/quality/phred63.hpp:42]:
```suggestion
 * \see quality, it contains a comprehensive overview over all Phred implementations and what a Phred score represents.
```

File: include/seqan3/alphabet/quality/phred94.hpp [hunk 10]
```diff
@@ -36,7 +36,7 @@ namespace seqan3
  *
  * \include test/snippet/alphabet/quality/phred94.cpp
  *
- * You can find more information about the Phred scores in our \ref quality submodule \see quality submodule.
+ * \see quality, it contains a comprehensive overview over all phred implementations and what a Phred score represents.
```
Comment (marehr) [inline include/seqan3/alphabet/quality/phred94.hpp:39]:
```suggestion
 * \see quality, it contains a comprehensive overview over all Phred implementations and what a Phred score represents.
```

**Review by marehr — COMMENTED** [review]
Just some small things and I think this is ready to merge :)

File: doc/tutorial/alphabet/index.md (hunk 4 above)
Comment (Irallia) [reply doc/tutorial/alphabet/index.md:189]: Yes it works without `\ref`. It looks like I didn't save my change.... sorry

**Review by marehr — APPROVED** [review]

**Pull request merged**